test/cpp/jit/test_ir.cpp

```diff
@@ -92,5 +92,88 @@ void testBlocks() {
       ->run(*g2);
 }
 
+void testCommonAncestor() {
+  std::string input_str = R"(
+graph(%x : Tensor,
+      %a.1 : bool,
+      %b.1 : bool,
+      %c.1 : bool):
+  %4 : int = prim::If(%a.1)
+    block0():
+      %5 : int = prim::If(%b.1)
+        block0():
+          %6 : int = prim::Constant[value=2]()
+          -> (%6)
+        block1():
+          %7 : int = prim::Constant[value=3]()
+          -> (%7)
+      -> (%5)
+    block1():
+      %8 : int = prim::If(%c.1)
+        block0():
+          %9 : int = prim::Constant[value=4]()
+          -> (%9)
+        block1():
+          %10 : int = prim::Constant[value=5]()
+          -> (%10)
+      -> (%8)
+  return (%4)
+)";
+
+  torch::jit::Graph g;
+  torch::jit::script::parseIR(input_str, &g);
+
+  std::vector<std::string> value_names{"6", "7", "9", "10"};
+  std::unordered_set<std::string> value_names_set(
+      value_names.begin(), value_names.end());
+  std::unordered_map<std::string, torch::jit::Value*> name_to_value;
+  std::function<void(Block*)> populate_values = [&](Block* b) {
+    for (Node* n : b->nodes()) {
+      if (n->outputs().size() == 1) {
+        auto name = n->output()->debugName();
+        if (value_names_set.count(name)) {
+          ASSERT_TRUE(!name_to_value.count(name));
+          name_to_value[name] = n->output();
+        }
+      }
+
+      for (Block* sub_b : n->blocks()) {
+        populate_values(sub_b);
+      }
+    }
+  };
+  populate_values(g.block());
+
+  ASSERT_EQ(name_to_value.size(), 4);
+
+  /* clang-format off */
+  int ref_blocks_from_graph[4][4] = {
```

Comment:
Encoding expected results like this works, another alternative is to include your previous algorithm in this test file and check against that.